mitodl/course-search-utils
# Allow using for UI management but not API mangement

### Relevant Issues

### What does this PR do?
This PR adds a few new exports:
1. `useSearchInputs` ... Provides state and UI handlers for search (facets, text, etc). This does not depend on search results at all. *In contrast, `useCourseSearch` does, since one of its inputs is the aggregations*
2. `useFacetOptions` ... which isolates the part of `useCourseSearch` that does depend on the search results, namely the aggregations data.
3. `useSyncUrlAndSearch` ... which syncs `searchParams` to URL and URL to `searchParams`.

See https://github.com/mitodl/open-discussions/pull/3936 for an example of these hooks in action.

I also refactored `useCourseSearch` a bit so that it uses both `useSearchInputs` and `useFacetOptions`. **`useCourseSearch` should not have any change in observable behavior.**

### How should/has this be manually tested?

This should be manually tested by: 
1. Checking out `cc/update-course-search-utils` in https://github.com/mitodl/ocw-hugo-themes/tree/cc/update-course-search-utils, running `yarn install`, `yarn start www`, and checking that http://localhost:3000/search/ works as expected:
    - search results appear to be correct, and update when you scroll
    - toggling a facet on/off automatically re-submits a query
    - edit the search text and click submit; query should be correct
    - check that the URL updates correctly when doing all of the above
    - try the back button in your browser. Search UI (facets, text) & and search results should update
    - check the network tab in dev console...Are extra search requests being made? _Should be "no"._

2. reviewing https://github.com/mitodl/open-discussions/pull/3936
